pkg/kubelet/kubelet_pods.go

```diff
@@ -175,7 +174,8 @@ func makeMounts(pod *v1.Pod, podDir string, container *v1.Container, hostName, h
 				return nil, cleanupAction, fmt.Errorf("unable to provision SubPath `%s`: %v", mount.SubPath, err)
 			}
 
-			fileinfo, err := os.Lstat(hostPath)
+			kubeletHostPath := mounter.KubeletPath(hostPath)
```

Comment:
Instead of exposing KubeletPath as an interface, would it be better to expose Lstat as an interface, and have EvalSymlinks return container path?

Reply:
Actually the code here is wrong. It resolves any symlinks in the `hostPath` in kubelet, which is not possible.

And that's Lstat is not easy - we need to resolve the symlinks on the host and then do `os.Lstat("/rootfs/" + resolved_path)` in kubelet

Reply:
Fixed by some refactoring.